src/Compilers/CSharp/Portable/Symbols/NamedTypeSymbol.cs

```diff
@@ -614,11 +617,20 @@ bool tryCalculateRequiredMembers(out ImmutableSegmentedDictionary<string, Symbol
                         }
                     }
 
+                    if (member.IsRequired())
+                    {
+                        // On the success path (which we want to optimize for), this member will need to be added to
+                        // requiredMembersBuilder. Add it before all checks are complete to ensure we only need to call
+                        // SymbolAsMember once.
+                        requiredMembersBuilder ??= baseAllRequiredMembers.ToBuilder();
+                        requiredMembersBuilder[member.Name] = member.SymbolAsMember(this);
```

Comment:
This is a behavioral change. Why would we not just capture `SymbolAsMember` to a local field and add it later if we need to? This is almost certainly the cause the test failure.

Reply:
Can you explain how this was a behavioral change?

Reply:
It seems like once we reach this point, one of three things happens:

1. `member` is not required, and it is never added to `requiredMembersBuilder`
2. `member` is required, and it is added to `requiredMembersBuilder`
3. `member` is required, but we return false before adding it to `requiredMembersBuilder`

Item (3) would be a behavior change _if_ `requiredMembersBuilder` was accessed when the return value is false, but the caller always just ignores it in that case.

Reply:
I misread the code here.